web/src/model/edit-actions.js

```diff
@@ -213,9 +213,32 @@ export const bufferRead = resource => dispatch => {
     });
 };
 
+// This is async because nested directories don't 
+// like opening until their parents have loaded
+export const directoryReadRecursive = resource => async dispatch => {
+  await resource.split('/').reduce(async (asyncUrl, asyncPart) => {
```

Comment:
Using the test/dev setup documented in the readme I was getting stack traces from React. If the starting URL was `http://localhost:3000` (the default for `yarn start`) then the `resource` value was undefined so the call to `split(...)` raised an exception. If starting from `http://localhost:3000/maiden` then things were okay.

I experimented with just wrapping the whole await expression inside `if (resource) { ... }` and that fixed it but that felt a tiny bit hacky to me... not sure what your thoughts are there.

Reply:
Ack. Yeah, I started with the `yarn start` dev server. When I tried serving straight from norns, the paths didn't match (`/` vs. `/maiden`, hence the try-catch), and that's when I started messing with the 404 code, but I didn't go back and try the dev server again. 

I don't think checking whether `resource` is truthy is too hacky, for my tastes at least. I'd prefer bailing if `!resource` over a big `if` block to reduce indentation, but that's well above the threshold of nitpickiness and I'd defer your preference on that.

Reply:
Did the check for `!resource`.